File: scripts/apitypings/testdata/genericmap/genericmap.ts
// From codersdk/genericmap.go
export interface Buzz {
  readonly foo: Foo
  readonly bazz: string
}

// From codersdk/genericmap.go
export interface Foo {
  readonly bar: string
}

// From codersdk/genericmap.go
export interface FooBuzz<R extends Custom> {
  readonly something: (readonly R[])
}

// From codersdk/genericmap.go
export type Custom = Foo | Buzz

File: scripts/apitypings/testdata/genericmap/genericmap.go
package codersdk

type Buzz struct {
	Foo  `json:"foo"`
	Bazz string `json:"bazz"`
}

type Foo struct {
	Bar string `json:"bar"`
}

type FooBuzz[R Custom] struct {
	Something []R `json:"something"`
}

type Custom interface {
	Foo | Buzz
}

// Not yet supported
//type FooBuzzMap[R Custom] struct {
//	Something map[string]R `json:"something"`
//}

// Not yet supported
//type FooBuzzAnonymousUnion[R Foo | Buzz] struct {
//	Something []R `json:"something"`
//}
